django/db/models/expressions.py

```diff
@@ -1504,14 +1533,7 @@ def as_sql(self, compiler, connection, **extra_context):
                 **extra_context,
             )
         except EmptyResultSet:
-            if self.negated:
-                features = compiler.connection.features
-                if not features.supports_boolean_expr_in_select_clause:
-                    return "1=1", ()
-                return compiler.compile(Value(True))
             raise
-        if self.negated:
-            sql = "NOT {}".format(sql)
```

Comment:
We should remove a similar workaround in `Exists()`.

Reply:
I think I'm missing something.
I can't figure out that `When()` is using a `negated` kwargs 🤔 
So I'm not sure what to change for the `When()` class.

Reply:
Nevermind, I was confused by diff.